### ISSUE TITLE
Compilation errors with arduino_esp32 version 3.0.0

### ISSUE THREAD

**Issue by mazgch** [gilmaimon/ArduinoWebsockets#156]
**Describe the bug**

```
In file included from /../ArduinoWebsockets/src/tiny_websockets/internals/ws_common.hpp:35,
                 from /../ArduinoWebsockets/src/tiny_websockets/message.hpp:3,
                 from /../ArduinoWebsockets/src/ArduinoWebsockets.h:4,
                 from /private/var/folders/gp/8jsyb2256k32d6sg4qyg25rr0000gn/T/.arduinoIDE-unsaved2024429-2129-qcvm11.y88m/Esp32-Server/Esp32-Server.ino:20:
/../ArduinoWebsockets/src/tiny_websockets/network/esp32/esp32_tcp.hpp:16:60: error: 'WiFiClientSecure' was not declared in this scope; did you mean 'WiFiClient'?
   16 |   class SecuredEsp32TcpClient : public GenericEspTcpClient<WiFiClientSecure> {
      |                                                            ^~~~~~~~~~~~~~~~
      |                                                            WiFiClient
/../ArduinoWebsockets/src/tiny_websockets/network/esp32/esp32_tcp.hpp:16:76: error: template argument 1 is invalid
   16 |   class SecuredEsp32TcpClient : public GenericEspTcpClient<WiFiClientSecure> {
      |                                                                            ^
/../ArduinoWebsockets/src/tiny_websockets/network/esp32/esp32_tcp.hpp: In member function 'void websockets::network::SecuredEsp32TcpClient::setCACert(const char*)':
/../ArduinoWebsockets/src/tiny_websockets/network/esp32/esp32_tcp.hpp:19:13: error: 'class websockets::network::SecuredEsp32TcpClient' has no member named 'client'
   19 |       this->client.setCACert(ca_cert);
      |             ^~~~~~
/..ArduinoWebsockets/src/tiny_websockets/network/esp32/esp32_tcp.hpp: In member function 'void websockets::network::SecuredEsp32TcpClient::setCertificate(const char*)':
/..ArduinoWebsockets/src/tiny_websockets/network/esp32/esp32_tcp.hpp:23:13: error: 'class websockets::network::SecuredEsp32TcpClient' has no member named 'client'
   23 |       this->client.setCertificate(client_ca);
      |             ^~~~~~
/../ArduinoWebsockets/src/tiny_websockets/network/esp32/esp32_tcp.hpp: In member function 'void websockets::network::SecuredEsp32TcpClient::setPrivateKey(const char*)':
/../ArduinoWebsockets/src/tiny_websockets/network/esp32/esp32_tcp.hpp:27:13: error: 'class websockets::network::SecuredEsp32TcpClient' has no member named 'client'
   27 |       this->client.setPrivateKey(private_key);
      |             ^~~~~~
```

**To Reproduce**
Install latest **arduino_esp32 version 3.0.0** https://github.com/espressif/arduino-esp32/releases/tag/3.0.0
Compile an ArduinoWebSocket example, e.g Esp32-Server.ion

**Expected behavior**
Compiles well

**Code**
no changes

**Additional context**
Install latest **arduino_esp32 version 3.0.0**

**Comment by mazgch** [comment]
seems a `#include <WiFiClientSecure.h>` may be needed in at least esp32_tcp.hpp

**Comment by Mukund2900** [comment]
I am getting the same issue. Which version needs to be used here ?

**Comment by erickcrus** [comment]
> seems a `#include <WiFiClientSecure.h>` may be needed in at least esp32_tcp.hpp

This works for me.

**Comment by gilmaimon** [comment]
Thanks @mazgch and everyone - I've released 0.5.4

Can you confirm it solves your issue?

**Comment by mazgch** [comment]
Solved by release 0.5.4, can be further improved by https://github.com/gilmaimon/ArduinoWebsockets/pull/160
